monitoring/storage.go

```diff
@@ -34,14 +36,43 @@ type StorageConfig struct {
 	Filesystems []string
 	// MinFreeBytes define minimum free volume capacity
 	MinFreeBytes uint64
-	// HighWatermark is the disk occupancy percentage that is considered degrading
-	HighWatermark uint
+	// WatermarkWarning is the disk occupancy percentage that will trigger a warning probe
+	WatermarkWarning uint
+	// WatermarkCritical is the disk occupancy percentage that will trigger a critical probe
+	WatermarkCritical uint
+}
+
+func (c *StorageConfig) CheckAndSetDefaults() error {
+	var errors []error
+	if c.Path == "" {
+		errors = append(errors, trace.BadParameter("volume path must be provided"))
+	}
+
+	if c.WatermarkWarning > 100 {
+		errors = append(errors, trace.BadParameter("warning watermark must be between 0 and 100"))
+	}
+
+	if c.WatermarkCritical > 100 {
+		errors = append(errors, trace.BadParameter("critical watermark must be between 0 and 100"))
```

Comment:
Do you want to make sure Critical > Warning?

Reply:
The ui might be a little weird if we ensure that critical > warning. Example: if we set critical to 50 and leave warning unset, warning will default to 80. 

I guess what we can do is first set defaults. Then, if warning > critical just set warning == critical?